decidim-api/lib/decidim/api/component_mutation_type.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Api
+    class ComponentMutationType < GraphQL::Schema::Union
+      description "A component mutation."
+
+      possible_types Decidim::Proposals::ProposalsMutationType # ,
+      #  ::Decidim::Api::Budgets::BudgetsMutationType,
+      #  ::Decidim::Api::Accountability::AccountabilityMutationType
+
+      def self.resolve_type(obj, _ctx)
+        case obj.manifest_name
+        when "proposals"
+          Decidim::Proposals::ProposalsMutationType
+          # when "budgets"
+          #   Decidim::Api::Budgets::BudgetsMutationType
+          # when "accountability"
+          #   Decidim::Api::Accountability::AccountabilityMutationType
+        end
+      end
```

Comment:
Can we make this dynamic so that we are not assuming any cross dependencies at the `decidim-api` module?

```suggestion
      def self.resolve_type(obj, _ctx)
        mod = obj.manifest_name.camelize
        "Decidim::#{mod}::#{mod}MutationType".constantize
      rescue NameError
        # The component mutation is not defined, type cannot be resolved.
      end
```